flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/util/OperatorInfo.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.graph.util;
+
+import org.apache.flink.api.common.operators.ResourceSpec;
+import org.apache.flink.streaming.api.graph.StreamConfig;
+import org.apache.flink.streaming.api.graph.StreamEdge;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** A public class to help maintain the information of an operator */
+public class OperatorInfo {
+
+    private String chainedName;
+
+    private ResourceSpec chainedMinResource;
+
+    private ResourceSpec chainedPreferredResources;
+
+    private StreamConfig vertexConfig;
+
+    /**
+     * The config for node chained with the current operator, the key is the ID of the stream node
+     */
+    private final Map<Integer, StreamConfig> chainedConfigs;
```

Comment:
Would it be better to move this field to OperatorChainInfo? The other chainedXXX fields are similar, include  chainedName, chainedMinResource and chainedPreferredResources.

Reply:
> Would it be better to move this field to OperatorChainInfo? The other chainedXXX fields are similar, include chainedName, chainedMinResource and chainedPreferredResources.

Thank you for your reminder. I have moved it to JobVertexBuildContext, which does not have a one-to-one correspondence with ChainInfo because in a multi input chain containing the source chain, the key may also be the id of all operators in that chain.